codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -299,6 +299,17 @@ public static boolean isTransferEncodingChunked(HttpMessage message) {
     }
 
     /**
+     * Checks to see if {@link HttpMessage} has empty content.
+     *
+     * @param message The message to check
+     * @return True if headers indicate empty content
+     */
+    public static boolean hasEmptyContent(HttpMessage message) {
+        return !message.headers().contains(HttpHeaderNames.TRANSFER_ENCODING)
+                && getContentLength(message, 0) == 0;
```

Comment:
@trustin doh you are right... I forgot about this. So I think the correct would be to check:
```java
&& getContentLength(message, -1) == 0;
```